src/function/chatPostMessage.js

```diff
@@ -0,0 +1,70 @@
+import https from "https";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+const TOKEN = process.env.LINE_ACCESS_TOKEN;
+
+const HEADERS = {
+	"Content-Type": "application/json",
+	Authorization: `Bearer ${TOKEN}`,
+};
+
+// Send four-day weather forecast information to user
+function chatPostMessage(replyToken, weatherForecastData, messageText) {
+	const weatherForecastForFourDays = [];
+
+	//Eight weather data is sent per day from weatherAPI(each data represents data for every three hours)
+	const weatherDataPerDay = 8;
+	//Five days of weather data is sent from weatherAPI(but this bot uses data for only four days)
+	const weatherDataPerFiveDays = weatherDataPerDay * 5;
+
+	for (let i = 0; i < weatherDataPerFiveDays - weatherDataPerDay; i += weatherDataPerDay) {
+		// Forecast date data contains time information such as YYYY-MM-DD HH:MM:SS(but this bot uses only YYYY-MM-DD)
+		//Extract 10 characters from left which is YYYY-MM-DD
+		const forecastDate = weatherForecastData.list[i].dt_txt.substring(0, 11);
+
+		const weatherDescription = weatherForecastData.list[i].weather[0].main;
+
+		const forecastObj = {
+			type: "text",
+			text: `${forecastDate}: ${weatherDescription}`,
+		};
+		weatherForecastForFourDays.push(forecastObj);
+	}
+
+	// Add initial descriptive text to message
+	const firstMessage = {
+		type: "text",
+		text: `This is the four-day weather forecast for ${messageText}.`,
+	};
+	weatherForecastForFourDays.unshift(firstMessage);
```

Comment:
Why do you use unshift here instead of put the data when weatherForecastForFourDays is instantiated?
```
const weatherForecastForFourDays = [{type: "text", text: `This is the four-day weather forecast for ${messageText}.` }];
```

Reply:
I had it out of weatherForecastForFourDays so I could see what this message is. But for the sake of code brevity I'll put it inside it.